### ISSUE TITLE
User friendly message for non-existing folder

### ISSUE THREAD

**Issue by CyrilleB79** [opensourcesys/speechLogger#12]
If I enter a non-exsisting path for the speech logger folder and validate the settings dialog, nothing happens.
Only the following error is logged:
```
ERROR - external:globalPlugins.speechLogger.GlobalPlugin.applyUserConfig (15:50:32.105) - MainThread (15320):
The folder given for log files does not exist (%tmp%\nvdaSpeechLogger).
```

Then the logging cannot occur and an error is logged in the log to indicate this.

When validating on an non-existing folder, I'd expect at least a warning message (dialog). Or better, an error message (dialog) preventing the settings to be validated as is.

If you want to support a use case where a non-existing path should still be supported, e.g. only log when an USB stick is present, you may opt for the warning solution though.

**Comment by XLTechie** [comment]
@CyrilleB79 What would you think, if I just made the directory edit field read-only, and required use of the browse button to enter directory? The UX is approximately the same.

**Comment by CyrilleB79** [comment]
That may be a good solution, provided you ensure that the browse folder dialog does not allow to enter a non-existing path.

**Comment by XLTechie** [comment]
Yes, it already does. I originally borrowed that logic from the create portable NVDA tool. I neglected to take into account, that if you enter any number of non-extant directories there in the main dialog (without hitting "browse"), NVDA will just silently create as many parent directories as it needs to make the folder valid. I didn't implement that logic for Speech Logger, and I don't think it would be good to do so. I'd rather disable the field.

**Comment by XLTechie** [comment]
I have a preliminary version of this solution in [V24.2.502](https://github.com/opensourcesys/speechLogger/releases/download/v24.2.502-dev/speechLogger-24.2.502.nvda-addon), currently processing to be in the dev channel of the store.

Unfortunately, it has one problem: the edit field is not in the tab/focus order. It appears not to be there at all, unless you use object nav. I'm sure the fix for this is easy, I just don't have time to deal with it tonight.

This practically solves the problem of this issue, though I consider it ugly in the current implementation.

**Comment by CyrilleB79** [comment]
If you want to make the field disabled (read-only) but readable, you probably need to copy what has been done for the Synthesizer field or the Braille display field in NVDA settings.

But finally, maybe it's easier for you to keep the field editable and to issue a warning when validating the dialog if the folder does not exist?

I am fine with either solution.

**Comment by XLTechie** [comment]
Yes, that's my intent (to see what is done for one of the other read-only but focusable fields). I am dealing with a dangerous mold problem from a slowly leaking pipe behind a wall in my home, so haven't had much time to look at this issue, but hopefully I'll finish it up tonight.

**Comment by CyrilleB79** [comment]
OK. There is no hurry.

I hope that you'll fix your leak at home... That's the top priority of course!

**Comment by XLTechie** [comment]
It looks like they use an expando for this purpose. I'm not sure I'll be able to do that, because the text field is generated inside the `gui.guiHelper.PathSelectionHelper` class with no field type customization possible. I have one other thing to try, but if that fails, as I think is likely, I will return to your original request of validating the unnecessary first path entry field.
